eessi/testsuite/hooks.py

```diff
@@ -282,16 +282,54 @@ def _assign_one_task_per_gpu(test: rfm.RegressionTest):
     log(f'num_cpus_per_task set to {test.num_cpus_per_task}')
     log(f'num_tasks set to {test.num_tasks}')
 
+def _set_or_append_valid_systems(test: rfm.RegressionTest, valid_systems: str):
+    """
+    Sets test.valid_systems based on the valid_systems argument.
+    - When valid_systems is an empty string, test.valid_systems will be set equal to eessi.testsuite.constants.INVALID_SYSTEM
+    - When no test.valid_system was set yet, or it was at the default value ['*'], it will be overwritten by [valid_system]
+    - When a test.valid_system was already set, valid_system will be appended to it. This allows adding requests for multiple partition features by different hooks.
+    """
+
+    # This indicates an invalid test that always has to be filtered
+    if valid_systems == '':
+        test.valid_systems = [INVALID_SYSTEM]
+        return
+
+    # test.valid_systems wasn't set yet, so set it
+    if len(test.valid_systems) == 0:
+        test.valid_systems = [valid_systems]
```

Comment:
so, my suggestion is to change it this:
```suggestion
    if len(test.valid_systems) == 0:
        # test.valid_systems is empty, which means all tests ar filtered out
        pass
```